cluster/images/etcd/migrate/util_others.go

```diff
@@ -0,0 +1,27 @@
+// +build !windows
+
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/mrunalp/fileutils"
+)
+
+func copyDirectory(source string, dest string) error {
+	return fileutils.CopyDirectory(source, dest)
```

Comment:
I'm not thrilled with the level of test coverage for this (none in https://github.com/mrunalp/fileutils)

Our unit test is pretty thin (just makes sure a file exists). @jpbetz, are there weird things in the data dir we need to handle specially (links, sockets, files with weird permissions, etc)?

Reply:
No links, sockets or strange permissions, fortunately. The data directory contains `.wal` files, `.snap` files, the main `db` file and some temporary `db.tmp` and `.snap.db` files for specific operations. They are all normal files.

Reply:
ok. does this produce files with the same permissions as `cp`? I don't know how much we rely on file permissions here to protect the etcd data

Reply:
some background, there's a reference in vendor/github.com/opencontainers/runc/libcontainer/rootfs_linux.go already so i picked this impl.